src/python/py/models/builder.py

```diff
@@ -331,15 +333,20 @@ def save_model(self, out_dir):
         )
 
     def to_int4(self, model):
-        quant = MatMul4BitsQuantizer(
-            model=model,
-            block_size=self.quant_attrs["int4"]["block_size"],
-            is_symmetric=True,
-            accuracy_level=self.quant_attrs["int4"]["accuracy_level"],
-            nodes_to_exclude=[],
-        )
-        quant.process()
-        return quant.model.model
+        if (self.quant_provider == "inc"):
+            quant = MatMul4BitsQuantizer(
+                model=model,
+                block_size=self.quant_attrs["int4"]["block_size"],
+                is_symmetric=True,
+                accuracy_level=self.quant_attrs["int4"]["accuracy_level"],
+                nodes_to_exclude=[],
+            )
+            quant.process()
+            return quant.model.model
+        else:
+            import modelopt
```

Comment:
Hi, I don't think the modelopt int4 quantization is compatible with the model builder. At least, not with the current state of things. 

The default int4 quantizer `MatMul4BitsQuantizer` uses an onnxruntime contrib op called `MatMulNBits` for int4 quantization. But the modelopt quantizes uses int4 DequantizeLinear and regular MatMul. 
However, onnxruntime currently doesn't support onnx int4 datatype so it cannot even load a model with int4 dq nodes.  Initial support for int4 to ort main was only merged recently https://github.com/microsoft/onnxruntime/pull/20362 and it only supports cpu ep. 

I think a more general approach might be to just use model builder to create a float32/float16 model, and then quantize using modelopt subsequently? Nvidia also contributed the nvmo quantization pass to Olive https://github.com/microsoft/Olive/blob/3d3277b646c2c5a0fa96bcf629a117aa78423176/olive/passes/onnx/nvmo_quantization.py#L47 so it is possible to create a workflow in Olive to first do model builder and then nvmo quantization if desired.

Reply:
Also, I think the modelopt quantizer requires calibration data but is not provided here?